RebornCore/src/main/java/reborncore/common/BaseBlockEntityProvider.java

```diff
@@ -49,27 +50,29 @@ public Optional<ItemStack> getDropWithContents(World world, BlockPos pos, ItemSt
 		if (blockEntity == null) {
 			return Optional.empty();
 		}
+
 		ItemStack newStack = stack.copy();
-		NbtCompound blockEntityData = blockEntity.createNbt();
-		stripLocationData(blockEntityData);
-		if (!newStack.hasNbt()) {
-			newStack.setNbt(new NbtCompound());
-		}
-		newStack.getOrCreateNbt().put("blockEntity_data", blockEntityData);
+		newStack.applyComponentsFrom(blockEntity.createComponentMap());
```

Comment:
This isnt correct, block entities still have none component data that you would want to capture here.

Reply:
Ah, just realised you likely strugged to know how to save the data to the stack. You will need to create a new custom component to store the Block's NBT. I expect we will have a handful of custom compoents for data that we used to store in NBT. Let me know if you want me to create one or two as an example.

Reply:
This is used to save inventory. Same is done in ShulkerBox. There is a Component for Inventory already.

Reply:
I think we want to store the entire block entity data, otherwise we would need to move all of our BEs to use components to store their inventory.